module-4/backend/week_two.md

```diff
@@ -5,17 +5,29 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What is Webpack and why is it useful?
+Webpack is a compiler that converts all JS to ES5 for browser compatability & compiles & minifies js, css, and html.
 2. When do you want to use event delegation?
+Event delegation is used to listen for events on a higher "bubble" than the actualy event; you can listen to something on the "outside" & delegate tasks according to the event's target.
 3. What's one difference between ES5 and ES6?
+ES6 phases out semicolons, uses arrow functions, implements "classes" (similar to constructor functions), uses `const` & `let` instead of `var` and allows parameters to have default values.
 4. How are you using the MVC design pattern in your Quantified Self project?
-5. How do you execute raw SQL in node?
+We're implementing models & controllers in files & directors structured similarly to rails. We extracted out SQL queries and other functions related to our "models" according to their responsibility and extracted the callback functions in our "routes" (from server.js) into controllers with functions named like controller actions
+5. How do you execute raw SQL in node? 
+`.raw(<SQL Query>)`
 6. What's a callback function and what are some reasons when we use/need callback functions?
+Callback functions are functions that are passed as parameters to other functions; we use them for traversing the dom (when an event occurs, something should happen), and with enumerables. We use it when a function really only needs to be run as the result of another function being called.
 7. What is CORS?
+CORS allows our front-end local host port to communicate with our back-end localhost port
 8. What are some steps to avoid CORS?
+Not sure; I thought we wanted to use CORS.
```

Comment:
CORS stands for Cross-Origin Resource Sharing and sometimes we get CORS errors if we are requesting something from another domain. if we do get CORS errors, we can do a few things to get around it - we can use a package or a gem to help us out